### ISSUE TITLE
Paint worklets: property shorthands

### ISSUE THREAD

**Issue by asajeffrey** [servo/servo#17378]
The spec for registering a new paint worklet (https://drafts.css-houdini.org/css-paint-api/#dom-paintworkletglobalscope-registerpaint) has a note "Note: The list of CSS properties may contain shorthands." We are not currently supporting this.

**Comment by asajeffrey** [comment]
This requires a notion of computed value and normalization for CSS shorthand properties, which should be based on typed CSSOM. The spec issue is https://github.com/w3c/css-houdini-drafts/issues/161

**Comment by nox** [comment]
Typed CSSOM isn't implemented, right?

**Comment by asajeffrey** [comment]
Yes, but this is an issue even in the absence of typed CSSOM, e.g. a worklet can monitor "background" and expect to be notified about changes to all the `background-*` properties. It's quite a drastic change for servo, since it means the declared-values to computed-values transform is bidirectional.

**Comment by nox** [comment]
What do you mean by the bidirectional part? What's 'declared-values'?

**Comment by emilio** [comment]
What's the problem of just mapping `background` to all its longhands? How does this differ from the mapping we need to do for stuff like `transition`?

**Comment by asajeffrey** [comment]
Step 7 of https://drafts.css-houdini.org/css-paint-api/#invoke-a-paint-callback says "Let styleMap be a new StylePropertyMapReadOnly populated with only the computed value’s for properties listed in inputProperties", which can include shorthands. So we need a way to provide a computed value for a shorthand.

@nox: I guess I was being sloppy, it's the shorthand to longhand map that is being made bidirectional.

@emilo: it's not just a question of concatenating the longhands is it? For example for background, if the user sets `background-image: url('foo'), url('bar')` and `background-position: left, right`, then the computed value for `background` should be `background: url('foo') left, url('bar') right`?

**Comment by asajeffrey** [comment]
@emilio: hmm, appending the shorthands does seem to be what's suggested vy https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-getpropertyvalue. That serializes the array of shorthands using https://drafts.csswg.org/cssom/#serialize-a-css-value, where step 8 says:

> Return the result of joining values as appropriate according to the grammar of shorthand and terminate these steps. 

I guess this is the bit I don't understand, and there's no hyperlink for me to follow! Do you know what's intended here?
